### DISCUSSION TITLE
Is there a comprehensive tutorial / usage example?

### DISCUSSION THREAD

**Discussion by marknefedov** [ory/kratos#3054] [Q&A] [answered]
Is there a tutorial or example project how to implement Ory stack in a project with a mobile app, spa, and an API? For example, some app where users can register and change their avatar and only logged-in users can see avatar for others API should return 403? Maybe something like what clerk did https://github.com/clerkinc/t3-turbo-and-clerk.

Also, I was reading a lot about ORY and other authentication solutions ([ZITADEL](https://zitadel.com/), [LogTo](https://logto.io/)). It looks like it is a common trend that authentication without webview or opening a browser on mobile is not possible, is it true with kratos too?

**Comment by kmherrmann** [comment]
For examples, please check our repository at https://github.com/ory/examples and the guides at https://www.ory.sh/docs/getting-started/integrate-auth/go.

Kratos and Ory Identities (cloud) support password-based native mobile login via our API flows, and we're adding support for passwordless and social sign-in in the near future, too!

**Comment by vinckr** [comment, marked as answer]
Hello @marknefedov 
AFAIK there is no 3 in one tutorial yet, we do have seperate tutorials for mobile, SPA and APIs going around. I do think though that something like the one you shared would be very compelling. Will look into how we can make that happen!

https://www.ory.sh/docs/getting-started/overview
https://github.com/ory/examples
There are also tutorials on our blog: https://www.ory.sh/blog/

Also on your last point, Ory Identities(Kratos) does not use OAuth2 for self-service flows, so no browser required on mobile! (social login on mobile is another story though)
See the react native tutorial here: https://www.ory.sh/login-react-native-authentication-example-api/

(edit: @kmherrmann was faster 😅 :)

**Reply by vinckr** [reply]
Followup @marknefedov 
I added Ory to the create-t3 template, and would see to add the expo/react native part to turbo next:
https://github.com/vinckr/create-t3-app-with-ory

**Reply by marknefedov** [reply]
Thank you very much!

**Reply by paulm17** [reply]
@vinckr I was just looking for this.  Thank you so much for posting it.

**Reply by vinckr** [reply]
Happy that it was of help @paulm17 !
If you have ideas how to improve it let me know. 
I wanted to extend it to the T3 Turbo example, but so far did not get around to it
